### PR TITLE
LevelManager and StartScreenTest

### PR DESCRIPTION
**Pull request by mlytvynchuk**
- Created a separate class for the level data manipulation.
- Added new dependencies in gradle.build.
- Created a StartScreenTest class.

### CODE REVIEW

File: app/src/main/java/com/google/droidjump/models/LevelManager.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    public static int getLevelsCount() {
+        return levelsCount;
+    }
+
+    public static void onCurrentLevelCompleted() {
+        int currentLevel = getCurrentLevel();
+        int lastLevel = getLastLevel();
+        if (currentLevel < levelsCount) {
+            // Increasing the current level
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/models/LevelManager.java:47]: nit:remove comment here and below.

File: app/src/main/java/com/google/droidjump/models/LevelManager.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import static com.google.droidjump.GameConstants.GAME_LEVELS_COUNT;
+import static com.google.droidjump.GameConstants.GAME_VIEW_CURRENT_LEVEL_STRING;
+import static com.google.droidjump.GameConstants.GAME_VIEW_DATA;
+import static com.google.droidjump.GameConstants.GAME_VIEW_LAST_LEVEL_STRING;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+public class LevelManager {
+    private static int levelsCount;
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/models/LevelManager.java:29]: this is kind of fixed. can we just use GAME_LEVELS_COUNT?

Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/models/LevelManager.java:29]: @redls GAME_LEVELS_COUNT is only a temporary solution. If I understand correctly, count of levels will be extracted from JSON configuration later. Is it right @Daniel-ND ?

File: app/src/main/java/com/google/droidjump/models/LevelManager.java (hunk 1 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/models/LevelManager.java]: Fixed.

File: app/src/main/java/com/google/droidjump/models/LevelManager.java (hunk 2 above)
Comment (Daniel-ND) [reply app/src/main/java/com/google/droidjump/models/LevelManager.java:29]: Yes, levelsCount will be based on data stored in config file

Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/models/LevelManager.java:29]: Added todo method for this purpose.

Comment (redls) [reply app/src/main/java/com/google/droidjump/models/LevelManager.java:29]: oh I see.

File: app/src/androidTest/java/com/google/droidjump/StartScreenTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    public ActivityTestRule<MainActivity> activityTestRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Before
+    public void setUp() throws Exception {
+        activityTestRule.launchActivity(MAIN_ACTIVITY_INTENT);
+    }
+
+    @Test
+    public void navigateToGameScreen() {
+        onView(ViewMatchers.withId(R.id.play_button)).perform(ViewActions.click());
```
Comment (saimonsaret) [inline app/src/androidTest/java/com/google/droidjump/StartScreenTest.java:50]: Does this test actually check something?

File: app/src/androidTest/java/com/google/droidjump/StartScreenTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    @Test
+    public void navigateToLevelsScreen() {
+        onView(ViewMatchers.withId(R.id.level_button)).perform(ViewActions.click());
+        onView(ViewMatchers.withId(R.id.choose_level_header)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void startNewGame() {
+        // Set the handpicked current level.
+        int testingCurrentLevel = GameConstants.GAME_LEVELS_COUNT / 2;
```
Comment (saimonsaret) [inline app/src/androidTest/java/com/google/droidjump/StartScreenTest.java:62]: Why do we need to divide it by 2?

Comment (mlytvynchuk) [reply app/src/androidTest/java/com/google/droidjump/StartScreenTest.java:62]: Please check next commit, I removed hardcoded values.

Comment (saimonsaret) [reply app/src/androidTest/java/com/google/droidjump/StartScreenTest.java:63]: That's good, but let's fix this PR as well if you are going to merge it.

File: app/src/androidTest/java/com/google/droidjump/StartScreenTest.java (hunk 3 above)
Comment (mlytvynchuk) [reply app/src/androidTest/java/com/google/droidjump/StartScreenTest.java:50]: Fixed.

File: app/src/androidTest/java/com/google/droidjump/StartScreenTest.java (hunk 4 above)
Comment (mlytvynchuk) [reply app/src/androidTest/java/com/google/droidjump/StartScreenTest.java:63]: Fixed.

**Pull request merged**